contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -2594,6 +2594,46 @@ SELECT
 WHERE FALSE;
 GRANT SELECT ON sys.spatial_index_tessellations TO PUBLIC;
 
+-- BABEL-3325: Revisit once DDL and/or CREATE EVENT NOTIFICATION is supported
+CREATE OR REPLACE VIEW sys.events 
+AS
+SELECT 
+  CAST(pt.tgfoid as int) AS object_id
+  , CAST(
+      CASE 
+        WHEN tr.event_manipulation='INSERT' THEN 1
+        WHEN tr.event_manipulation='UPDATE' THEN 2
+        WHEN tr.event_manipulation='DELETE' THEN 3
+        ELSE 1
+      END as int
+  ) AS type
+  , CAST(tr.event_manipulation as sys.nvarchar(60)) AS type_desc
+  , CAST(1 as sys.bit) AS  is_trigger_event
+  , CAST(null as int) AS event_group_type
+  , CAST(null as sys.nvarchar(60)) AS event_group_type_desc
+FROM information_schema.triggers tr
+JOIN pg_catalog.pg_namespace np ON tr.event_object_schema = np.nspname
+JOIN pg_class pc ON pc.relname = tr.event_object_table AND pc.relnamespace = np.oid
+JOIN pg_trigger pt ON pt.tgrelid = pc.oid AND tr.trigger_name = pt.tgname
+AND has_schema_privilege(pc.relnamespace, 'USAGE')
+AND has_table_privilege(pc.oid, 'SELECT,INSERT,UPDATE,DELETE,TRUNCATE,TRIGGER');
+GRANT SELECT ON sys.events TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.trigger_events
+AS
+SELECT
+  CAST(e.object_id as int) AS object_id,
+  CAST(e.type as int) AS type,
+  CAST(e.type_desc as sys.nvarchar(60)) AS type_desc,
+  CAST(0 as sys.bit) AS is_first,
+  CAST(0 as sys.bit) AS is_last,
+  CAST(null as int) AS event_group_type,
+  CAST(null as sys.nvarchar(60)) AS event_group_type_desc,
+  CAST(e.is_trigger_event as sys.bit) AS is_trigger_event
```

Comment:
I guess this field is not in the official document, but actually it exists, right?

Reply:
It is part of the [official document](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/system-catalog-views/sys-trigger-events-transact-sql.md#systrigger_events-transact-sql). `is_trigger_event` field is part of the columns inherited from `sys.events`

The reason I ordered the columns this way, with  `is_trigger_event` field being last, is to match the column ordering I saw in SQL Server when `sys.trigger_events` was queried: `select * from sys.trigger_events`.

Reply:
> Inherits the object_id, type, type_desc columns from [sys.events](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/relational-databases/system-catalog-views/sys-events-transact-sql.md).

Think the official document is not correct :)

Reply:
Ha, yes you're right the document seems to not be accurate. Unless directed otherwise, I'll keep the `is_trigger_event ` in for now. I tested on both SQL server 2014 & 2019, when I query `sys.trigger_events` it has the `is_trigger_event` column.